app/backend/src/couchers/tasks.py

```diff
@@ -300,3 +300,19 @@ def enforce_community_memberships():
                     )
                 )
             session.commit()
+
+
+def enforce_community_memberships_for_user(session, user):
+    """
+    Adds a given user to all the communities they belong in based on their location.
+    """
+    nodes = session.execute(select(Node).where(func.ST_Contains(Node.geom, user.geom))).scalars().all()
+    for node in nodes:
+        logger.info(node.id)
+        node.official_cluster.cluster_subscriptions.append(
+            ClusterSubscription(
+                user=user,
+                role=ClusterRole.member,
+            )
+        )
+    session.commit()
```

Comment:
Is there a way to combine this with `enforce_community_memberships`? For maintainability, I think it would be good to add all users to communities with the same method.

Alternatively, implement a test to make sure that both work the same (i.e. each adds one user to the same community)

Reply:
> Is there a way to combine this with `enforce_community_memberships`? For maintainability, I think it would be good to add all users to communities with the same method.

We kind of could, I suppose. The other one is `for each node ... for each user not yet added ...`, whereas this is `for each node ...`, so there's not really any way to run this without running for each user (regardless of if added or not), so it'd do a tonne of extra work.

Reply:
You could pull out the guts and have nodes and users as input. Or even just users

Reply:
I think it's good as is; they're right next to each other and it's pretty easy to maintain. Our tests basically verify that both are correct.